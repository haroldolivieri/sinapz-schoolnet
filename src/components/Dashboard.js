import React from 'react';
import Grid from "@material-ui/core/Grid";
import { DashboardCard, DashboardPaper } from '.'
import PropTypes from 'prop-types'

export const DashboardIndice = ({
    quantidade = 0,
    tipo = 'Unknown',
}) => (
    <Grid item md={4}>
        <DashboardCard
            // backgroundColor="#00f"
            quantidade = { quantidade }
            tipo = { tipo }
        />
    </Grid>
)
DashboardIndice.propTypes = {
    quantidade: PropTypes.number,
    tipo: PropTypes.string,
}

export const DashboardAvaliacao = ({
    tipo = 'Unknown',
    disciplina = 'Unknown',
    grupo = 'Unknown',
    tempoRestante = 'Unknown'
}) => (
    <DashboardPaper
        tipo = { tipo }
        disciplina = { disciplina }
        grupo = { grupo}
        tempoRestante = { tempoRestante }
    />
)
DashboardAvaliacao.propTypes = {
    tipo: PropTypes.string,
    disciplina: PropTypes.string,
    grupo: PropTypes.string,
    tempoRestante: PropTypes.string,
}

const Dashboard = () => (
    <div>
        <div>
            <Grid container spacing={24}>
            {/* Start DashboardIndices Temp */}
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "5"
                        tipo = "Filiais"
                    />
                </Grid>
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "40"
                        tipo = "Professores"
                    />
                </Grid>
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "200"
                        tipo = "Alunos"
                    />
                </Grid>
            {/* End Temp */}
            {/* {DashboardIndices.map(item => (
                    <DashboardIndice
                        quantidade={item.quantidade}
                        tipo={item.tipo}
                    />
                ))} */}
            </Grid>
        </div>
        <div>
            <h1>Próximas Avaliações em 24h </h1>
            {/* Start DashboardAvaliacoes Temp */}
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "30m" }
            />
            <br/>
            <DashboardPaper
                tipo={"Teste"}
                disciplina={ "Matematica" }
                grupo = { "Ensino Fundamental"}
                tempoRestante={ "1h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "2h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "30m" }
            />
            <br/>
            <DashboardPaper
                tipo={"Teste"}
                disciplina={ "Matematica" }
                grupo = { "Ensino Fundamental"}
                tempoRestante={ "1h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "2h" }
            />
            <br/>
            {/* End DashboardAvaliacoes Temp */}
            {/* {DashboardAvaliacoes.map(item => (
                    <DashboardAvaliacao
                        tipo={item.tipo}
                        disciplina={item.disciplina}
                        grupo={item.grupo}
                        tempoRestante={item.tempoRestante}
                    />
                ))} */}
        </div>
    </div>
)

export default Dashboard
