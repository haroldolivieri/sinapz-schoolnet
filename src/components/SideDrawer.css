.drawer {
    position: relative;
    z-index: 9999;
    height: 100vh;
    width: 80vw;
    background-color: #3C4754;
}

@media only screen and (min-width: 768px) {
    .drawer { 
        width: 30vw;   
    }
}