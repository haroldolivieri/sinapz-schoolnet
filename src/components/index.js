export { default as NotFound } from './NotFound'
export * from './NotImplementedYet'
export { default as ResponsiveDrawer } from './ResponsiveDrawer'
export { default as SideDrawer } from './SideDrawer'
export { default as Dashboard } from '../containers/Dashboard'
export { default as DashboardCard } from './DashboardCard'
export { default as DashboardPaper } from './DashboardPaper'
