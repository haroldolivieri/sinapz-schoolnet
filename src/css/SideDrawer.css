.drawer {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
    z-index: 9999;
    height: 100vh;
    width: 70vw;
    background-color: #3C4754;
}
@media only screen and (min-width: 900px) {
    .drawer { 
        width: 20vw;   
    }
}

.avatar {
    border-radius: 50%;
    width:8em;
    height:8em;
    margin: 1.5em;
}

.title {
    color: white;
    font-size: 1.2em;
    font-weight: 600;
    margin-bottom: 1em;
}

.list {
    width: 100%;
}

.item-text {
    font-size: 0.8em;
    color: white;
}