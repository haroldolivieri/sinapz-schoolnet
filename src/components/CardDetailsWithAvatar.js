import React from 'react'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import Avatar from '@material-ui/core/Avatar'
import '../css/CardDetailsWithAvatar.css'
import PropTypes from 'prop-types'
import { extractInitials } from '../util/handlerWithNames'

export const CardDetailsWithAvatar = ({branch}) => {
  return (
    <Card key={branch.id} className="card-container">
      <div className="avatar-item">
        <CardHeader avatar={
          <Avatar>{ extractInitials(branch.name) }</Avatar>
        }
        />
      </div>
      <div className="content-item">
        <span className="item-card-text-name">{branch.name}</span>
        <span className="item-card-text-sub">{`Respons√°vel: ${branch.inCharge}`}</span>
      </div>
    </Card>
  )
}

CardDetailsWithAvatar.propTypes = {
  branch: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    inCharge: PropTypes.string
  })
}

CardDetailsWithAvatar.defaultProps = {
  branch: {
    id: "Unknown",
    name: "Unknown",
    inCharge: "Unknown"
  }
}

export default CardDetailsWithAvatar