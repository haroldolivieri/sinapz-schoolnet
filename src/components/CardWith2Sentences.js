import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import '../css/CardWith2Sentences.css'

export const CardWith2Sentences = ({
                            FirstSentence = [<b className='highlight'>Hello</b>],
                            SecondSentence = [<b>]World!</b>],
}) => (
        <Card class="card">
            <CardContent  class="card_content">
                { FirstSentence }
                { SecondSentence }
            </CardContent>
        </Card>
);

CardWith2Sentences.propTypes = {
    FirstSentence: PropTypes.string,
    SecondSentence: PropTypes.string,
};