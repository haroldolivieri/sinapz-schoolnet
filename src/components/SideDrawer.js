import React from 'react'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import '../css/SideDrawer.css'

export const DrawerItem = ({
    text = 'Unknown',
    icon = <ItemIcon />,
    action = function() {}
}) => (
    <ListItem id="list-item" button key={text} onClick={action}>
        <ListItemIcon>{icon}</ListItemIcon>
        <p className="item-text">{text}</p>
    </ListItem>
)
DrawerItem.propTypes = {
    text: PropTypes.string,
    icon: PropTypes.element,
    action: PropTypes.func
}

export const ItemIcon = ({ img = 'undefined.svg', alt = '' }) => (
    <img src={require(`../images/${img}`)} alt={alt} />
)

export const SideDrawer = ({
    history,
    menuItems = [],
    schoolnetInfo = {
        name: 'Anonymous',
        photo:
            'https://cdn.shopify.com/s/files/1/0539/4361/products/anonymous-guy-fawkes-anarchy-decal.jpg?v=1410346765'
    }
}) => {
    const handleItemSelected = pathTo => {
        history.push(pathTo)
    }

    const AvatarImage = ({ url, alt = '' }) => (
        <img src={url} className="avatar" alt={alt} />
    )

    return (
        <div className="drawer">
            <AvatarImage url={schoolnetInfo.photo} alt={schoolnetInfo.name} />
            <span className="title">{schoolnetInfo.name}</span>
            <div className="list">
                {menuItems.map(item => (
                    <DrawerItem
                        id="drawer-item"
                        key={item.name}
                        text={item.name}
                        icon={<ItemIcon img={item.icon} alt={item.name} />}
                        action={handleItemSelected.bind(this, item.path)}
                    />
                ))}
            </div>
        </div>
    )
}

const mapStateToProps = state => ({
    schoolnetInfo: state.menu.get('schoolnetInfo'),
    menuItems: state.menu.get('menuItems')
})

export default compose(
    withRouter,
    connect(mapStateToProps)
)(SideDrawer)

SideDrawer.propTypes = {
    history: PropTypes.object,
    menuItems: PropTypes.array,
    schoolnetInfo: PropTypes.object
}
