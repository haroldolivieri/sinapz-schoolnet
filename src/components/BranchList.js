import React from 'react'
import CardDetailsWithAvatar from './CardDetailsWithAvatar'

const BranchList = ({ branches }) => {
  if(!branches || !branches.length){
    return <p>Essa rede n√£o tem filiais</p>
  }
  return branches.map(branch => {
    return(
      <CardDetailsWithAvatar branch={branch} key={branch.id} />
    )
  })
}

export default BranchList
