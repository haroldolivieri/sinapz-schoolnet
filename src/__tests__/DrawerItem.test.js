import React from 'react';
import { DrawerItem } from '../components/SideDrawer';
import { shallow } from 'enzyme';

describe('Drawer Item', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<DrawerItem/>)
        expect(wrapper).toMatchSnapshot()
    });

    it('call right action method when item is clicked', () => {
        const onClickMock = jest.fn();
        const wrapper = shallow(
            <DrawerItem action={onClickMock}/>,
        );

        wrapper.find('#list-item').simulate('click');

        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});