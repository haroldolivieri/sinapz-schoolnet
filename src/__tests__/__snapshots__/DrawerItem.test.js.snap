// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Item renders correctly 1`] = `
<WithStyles(ListItem)
  button={true}
  id="list-item"
  key="Unknown"
>
  <WithStyles(ListItemIcon)>
    <pure(MoveToInboxIcon) />
  </WithStyles(ListItemIcon)>
  <WithStyles(ListItemText)
    primary="Unknown"
  />
</WithStyles(ListItem)>
`;
